utils/ignore_files.go

```
@@ -0,0 +1,52 @@
+package utils
+
+import (
+	"bytes"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+func GetIgnorefiles(fs afero.Fs, workingDirectoryPath string, ignorefilePath string) ([]string, error) {
+	ignoreFileExists, err := afero.Exists(fs, ignorefilePath)
+	if err != nil {
+		return nil, err
+	}
+	ignorefiles := []string{}
+	if ignoreFileExists {
+		ignoreFile, err := afero.ReadFile(fs, ignorefilePath)
+		if err != nil {
+			return nil, err
+		}
+		n := bytes.IndexByte(ignoreFile, 0)
+		if n > 0 {
+			ignorefiles = strings.Split(string(ignoreFile[:n]), "/n")
+		}
+		for i, ignorefile := range ignorefiles {
+			ignorefiles[i] = filepath.Join(workingDirectoryPath, ignorefile)
+		}
+	}
+	return ignorefiles, err
```

I don't know if architect uses masked errors. I see it doesn't in `workflow.go`.